src/core/geometry/qgsgeometry.cpp

```diff
@@ -171,6 +171,16 @@ QgsGeometry QgsGeometry::fromPointXY( const QgsPointXY &point )
   return QgsGeometry();
 }
 
+QgsGeometry QgsGeometry::fromPoint( const QgsPoint &point )
+{
+  std::unique_ptr< QgsAbstractGeometry > geom( QgsGeometryFactory::fromPoint( point ) );
+  if ( geom )
+  {
+    return QgsGeometry( geom.release() );
+  }
+  return QgsGeometry();
```

Comment:
```suggestion
  return QgsGeometry( point.clone() );
```

and remove the unnecessary QgsGeometryFactory::fromPoint function.

Reply:
Done.